### DISCUSSION TITLE
Can I `-m config` for a certain `--display` rather than a `--space`?

### DISCUSSION THREAD

**Discussion by fiedl** [asmvik/yabai#959] [Q&A] [answered]
Hi, I have a multi-monitor setup and would like to have the `layout bsp` for two displays, `layout float` for one of them. Each display has several spaces.

I've seen

```
yabai -m config --space 5 layout bsp
```

in the [documentation](https://github.com/koekeishiya/yabai/blob/master/doc/yabai.asciidoc#6-domains).

Can I somehow do

```
yabai -m config --display 2 layout bsp
```

instead or maybe something equivalent?

**Comment by asmvik** [comment, marked as answer]
This is not possible using that specific syntax, although you can use the query system to retrieve a list of all spaces that belong to e.g display 2; the config is just a shell script.

**Reply by fiedl** [reply]
Thanks for the suggestion. This has also been my first thought on the search for a workaround; I just didn't know how to re-run the configuration when a display or a space is added or removed. Yabai's [signals](https://github.com/koekeishiya/yabai/blob/master/doc/yabai.asciidoc#signal) came to the rescue. This is what I ended up with:

```bash
# ~/.yabairc
# ...

for event in "mission_control_exit" "display_added" "display_removed" "display_moved"; do
  yabai -m signal --add event=$event action="~/.yabai_configure_space_layouts.rb"
done
```

```ruby
# ~/.yabai_configure_space_layouts.rb

#!/usr/bin/env ruby

# This script configures yabai to manage the windows on the secondary
# displays, but leave the windows on the primary display floating.
#
# See also: https://github.com/koekeishiya/yabai/discussions/959

require 'json'

displays = JSON.parse `yabai -m query --displays`
# [{
#   "id":2077751158,
#   "uuid":"FD6E9053-53B7-5224-5892-F9F7EC52CEF3",
#   "index":1,
#   "spaces":[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12],
#   "frame":{
#     "x":0.0000,
#     "y":0.0000,
#     "w":1680.0000,
#     "h":1050.0000
#   }
# }]

spaces_on_primary_display = displays.first['spaces']
spaces_on_secondary_displays = displays.drop(1).collect { |display| display['spaces'] }.flatten

for space in spaces_on_primary_display
  `yabai -m config --space #{space} layout float`
end

for space in spaces_on_secondary_displays
  `yabai -m config --space #{space} layout bsp`
end
```

This configures yabai to manage the windows on the secondary displays, but leave the windows on the primary display floating.
